src/main/kotlin/org/rust/lang/core/stubs/StubImplementations.kt

```diff
@@ -1112,8 +1112,8 @@ sealed class RsStubLiteralType(val typeOrdinal: Int) {
                     0 -> RsStubLiteralType.Boolean
                     1 -> RsStubLiteralType.Char(readBoolean())
                     2 -> RsStubLiteralType.String(readLong(), readBoolean())
-                    3 -> RsStubLiteralType.Integer(readEnumAsNullable<TyInteger.Kind>())
-                    4 -> RsStubLiteralType.Float(readEnumAsNullable<TyFloat.Kind>())
+                    3 -> RsStubLiteralType.Integer(readByte().toInt().takeIf { it >= 0 }?.let { TyInteger.VALUES[it] })
```

Comment:
Let's use `List.getOrNull`
I think `TyInteger.VALUES.getOrNull(readByte().toInt())` is more readable than `readByte().toInt().takeIf { it >= 0 }?.let { TyInteger.VALUES[it] }`